plugins/syslog/syslog_plugin.c

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2020, Open Mobile Platform LLC
+ */
+
+#include <stddef.h>
+#include <syslog.h>
+#include <tee_plugin_method.h>
+
+/*
+ * OPTEE has access to the plugin by the UUID
+ */
+#define SYSLOG_PLUGIN_UUID { 0x96bcf744, 0x4f72, 0x4866, \
+		{ 0xbf, 0x1d, 0x86, 0x34, 0xfd, 0x9c, 0x65, 0xe5 } }
+
+/* plugin cmd */
+#define TO_SYSLOG 0
+
+static TEEC_Result syslog_plugin_init(void)
+{
+	return TEEC_SUCCESS;
+}
+
+static TEEC_Result write_syslog(unsigned int sub_cmd, void *data, size_t data_len)
+{
+	(void)data_len;
+
+	/* 'sub_cmd' in this case means priority according syslog.h */
+	openlog(NULL, LOG_CONS | LOG_PID, LOG_DAEMON);
+	syslog((int)sub_cmd, "%s", (const char *)data);
```

Comment:
What if `data` isn't null terminated?

Reply:
In this example I use null-terminated strings for the plugin to pass them to `syslog()`.
But actually, `data_len` stores real size of the data. that's why we can use `data` and  `data_len` in any other cases of the use of plugins.

Reply:
Yes, obviously it works but it isn't robust. Bear in mind that people might copy these examples.
I'd prefer something more safe like:
``` C
syslog(sub_cmd, "%*s", (int)data_len, (const char *)data);
```

Reply:
Done